cpp/FileTransferAgent.cpp

```diff
@@ -610,21 +627,56 @@ void Snowflake::Client::FileTransferAgent::download(string *command)
     for (size_t i=0; i<m_largeFilesMeta.size(); i++)
     {
       m_executionResults->SetFileMetadata(&m_largeFilesMeta[i], i);
+      CXX_LOG_DEBUG("Putget serial large file download, %s file",
+                    m_largeFilesMeta[i].srcFileName.c_str());
+      if (isGetFastFailEnabled() && !m_failedTransfers.empty())
+      {
+        m_executionResults->SetTransferOutCome(RemoteStorageRequestOutcome::FAILED, i);
+        CXX_LOG_DEBUG("Sequential download, get fast fail enabled, Skipping file.");
+        continue;
```

Comment:
when fast fail is enabled, why do we need to continue here? These two conditions isGetFastFailEnabled && m_failedTransfers do not look like will change?

Reply:
Copied the same behavior from uploadFilesInParallel(). Continue to set transfer outcome for all following files and also have log for each file so we know which files are downloaded, which one failed and which are skipped afterwards.